vlm/src/main/scala/WarpRasterSource.scala

```diff
@@ -0,0 +1,58 @@
+package geotrellis.contrib.vlm
+
+import geotrellis.proj4._
+import geotrellis.raster._
+import geotrellis.raster.reproject.{ReprojectRasterExtent, Reproject}
+import geotrellis.vector._
+
+import java.net.URI
+
+
+class WarpRasterSource(
+  val base: RasterSource,
+  val crs: CRS,
+  val options: Reproject.Options
+) extends RasterSource {
+
+  def uri: URI = base.uri
+  def cols: Int = base.cols
+  def rows: Int = base.rows
+  def cellType: CellType = base.cellType
+  def bandCount: Int = base.bandCount
+
+  def extent: Extent = {
+    val oldProjectedExtent = ProjectedExtent(base.extent, base.crs)
+
+    oldProjectedExtent.reprojectAsPolygon(crs).envelope
+  }
+
+  private val transform = Transform(base.crs, crs)
+  private val backTransform = Transform(crs, base.crs)
+
+  def read(windows: Traversable[RasterExtent]): Iterator[Raster[MultibandTile]] = {
+    val rasterExtents: Traversable[RasterExtent] =
+      windows.map { case targetRasterExtent =>
+        val projectedExtent = ProjectedExtent(targetRasterExtent.extent, crs)
+
+        val sourceExtent: Extent = projectedExtent.reprojectAsPolygon(base.crs).envelope
+
+        RasterExtent(sourceExtent, targetRasterExtent.cols, targetRasterExtent.rows)
+      }
+
+    base.read(rasterExtents).map { raster =>
+      raster.reproject(base.crs, crs, options)
```

Comment:
`options` should not be passed here. It has parameters that control how the selection of the target `RasterExtent` is controlled. Basically in this `reproject` call `errorThreshold`, `parentGridExtent`, and `targetRasterExtent`, `targetCellSize` can all produces different `targetRasterExtent`.

Instead of `ResampleMethod` should be exposed and the version of reproject that uses `targetRaterExtent` should be used.

Reply:
I think that `reproject` method should be fine since we're replacing the `options` parameter with the `resampleMethod`. By default, `targetRasterExtent`, `parentGridExtent`, and `targetCellSize` are all `None` https://github.com/locationtech/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/reproject/Reproject.scala#L38-L40 So the only thing that's going to be set is the `method`.

In addition, the version of the `reproject` method that does take a `targetRasterExtent` will use the default `Options` if none are passed in https://github.com/locationtech/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/reproject/RasterReprojectMethods.scala#L28 So I don't see how that version is better.